app/controllers/mixins/checked_id_mixin.rb

```diff
@@ -117,6 +141,51 @@ def checked_or_params_id
       obj
     end
 
+
+    # Find a record by model and id and test it with RBAC
+    # Params:
+    #   klass   - class of accessed objects
+    #   ids     - accessed object ids
+    #   options - :with_flash : set flash in case user is not authorized,
+    # TODO:       :named_scope :
+    #
+    # Returns:
+    #   Array of selected class instances. If user does not have rights for it,
+    #   either sets flash or raises exception
+    #
+    def find_records_with_rbac(klass, ids, options = {})
+      ids ||= checked_or_params
+      filtered = Rbac.filtered(klass.where(:id => ids),
+                               :user => current_user,
+                               :named_scope => options[:named_scope])
+      unless ids.length == filtered.length
+        unauthorized_record = (ids - filtered.map { |record| record.id }).first
+        if options[:with_flash]
```

Comment:
Generally the message about record not being found and user not being authorized to view a particular record should be the same.

Possible attacker should not be able to use the differing error messages to make a list of entities existing in the system.

In some places in the app we display error display different error messages and in some places it may be even part of tests (even the QE tests) so we need to get rid of that.